include/multipass/platform.h

```diff
@@ -91,7 +91,11 @@ std::unique_ptr<Process> make_sshfs_server_process(const SSHFSServerConfig& conf
 std::unique_ptr<Process> make_process(std::unique_ptr<ProcessSpec>&& process_spec);
 int symlink_attr_from(const char* path, sftp_attributes_struct* attr);
 
-std::function<int()> make_quit_watchdog(); // call while single-threaded; call result later, in dedicated thread
+// Creates a function that will wait for signals or until the passed function returns false.
+// The passed function is checked every timeout milliseconds.
+// If a signal is received the optional contains it, otherwise the optional is empty.
+std::function<std::optional<int>(const std::function<bool()>&)> make_quit_watchdog(
+    const std::chrono::milliseconds& timeout); // call while single-threaded; call result later, in dedicated thread
```

Comment:
This parameter is used as a timeout in the implementation, but it's really a period for the caller, right? Should the name reflect that?

Reply:
Good point!